### PR TITLE
getmail6 6.18.12 (new formula)

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

replacement for python2 version of getmail
package inclusion, https://repology.org/project/getmail6/versions

### CODE REVIEW

File: Formula/getmail6.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  license "GPL-2.0-only"
+
+  depends_on "python@3.11"
+
+  def install
+    files = %w[getmail getmail_fetch getmail_maildir getmail_mbox]
+    libexec.install files
+    files.each do |file|
+      rewrite_shebang detected_python_shebang, libexec/file
+    end
```
Comment (carlocab) [inline Formula/getmail6.rb:17]:
```suggestion
    rewrite_shebang detected_python_shebang, *files
    libexec.install files
```

File: Formula/getmail6.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+class Getmail6 < Formula
+  include Language::Python::Shebang
+
+  desc "Extensible mail retrieval system with POP3, IMAP4, SSL support"
+  homepage "https://github.com/getmail6/getmail6"
+  url "https://github.com/getmail6/getmail6/archive/refs/tags/v6.18.12.tar.gz"
+  sha256 "419dc11184b147ef4daac2ec9f136b1b37e6450e2f44099629a6b7650deb044a"
+  license "GPL-2.0-only"
+
+  depends_on "python@3.11"
```
Comment (carlocab) [inline Formula/getmail6.rb:10]: Does it need a specific version of Python?

Comment (chenrui333) [reply Formula/getmail6.rb:10]: probably not, but I am specifying it mostly for being explicit.

File: Formula/getmail6.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
```
Comment (carlocab) [inline Formula/getmail6.rb:10]:
```suggestion
  uses_from_macos "python", since: :catalina
```

File: Formula/getmail6.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  def install
+    files = %w[getmail getmail_fetch getmail_maildir getmail_mbox]
+    rewrite_shebang detected_python_shebang, *files
```
Comment (carlocab) [inline Formula/getmail6.rb:14]:
```suggestion
    rewrite_shebang detected_python_shebang(use_python_from_path: true), *files
```

File: Formula/getmail6.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  sha256 "419dc11184b147ef4daac2ec9f136b1b37e6450e2f44099629a6b7650deb044a"
+  license "GPL-2.0-only"
+
+  uses_from_macos "python", since: :catalina
+
+  def install
+    files = %w[getmail getmail_fetch getmail_maildir getmail_mbox]
+    rewrite_shebang detected_python_shebang(use_python_from_path: true), *files
+    libexec.install files
+    bin.install_symlink Dir["#{libexec}/*"]
```
Comment (carlocab) [inline Formula/getmail6.rb:16]:
```suggestion
    bin.install_symlink libexec.children
```

File: Formula/getmail6.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    libexec.install "getmailcore"
+    man1.install Dir["docs/*.1"]
```
Comment (carlocab) [inline Formula/getmail6.rb:18]:
```suggestion
    man1.install Pathname.glob("docs/*.1")
```
These aren't directories, so let's avoid `Dir.[]`.

File: Formula/getmail6.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    files = %w[getmail getmail_fetch getmail_maildir getmail_mbox]
+    rewrite_shebang detected_python_shebang(use_python_from_path: true), *files
+    libexec.install files
+    bin.install_symlink libexec.children
+    libexec.install "getmailcore"
+    man1.install Pathname.glob("docs/*.1")
+  end
+
+  test do
+    system bin/"getmail", "--help"
```
Comment (SMillerDev) [inline Formula/getmail6.rb:22]: No tests?

Comment (chenrui333) [reply Formula/getmail6.rb]: make sense, just created a complete test for getmail6.

**Review by carlocab — APPROVED** [review]

**Pull request merged**